### PR TITLE
add sam sdk examples

### PR DESCRIPTION
**Pull request by shubhamiit**
# Description

# Checklist

- [x] I have read the contribution guidelines.
  - [General](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
  - [SDK](https://github.com/Azure/azureml-examples/blob/main/sdk/CONTRIBUTING.md)
  - [CLI](https://github.com/Azure/azureml-examples/blob/main/cli/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [x] Pull request includes test coverage for the included changes.
- [x] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure/azureml-examples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+    "      \"image\",\n",
+    "      \"input_points\",\n",
+    "      \"input_boxes\",\n",
+    "      \"input_labels\",\n",
+    "      \"multimask_output\"\n",
+    "    ],\n",
+    "        \"index\": [0,1,2,3,4],\n",
+    "        \"data\": [\n",
+    "            # segmentation per input point\n",
+    "            [base64.encodebytes(read_image(sample_image)).decode(\"utf-8\"),\"[[[[280,320]], [[300,350]]]]\",\"\", \"\", False],\n",
```
Comment (MadhuM02) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb:316]: nit: comment says single input, example is same as second example

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "dri",
```
Comment (MadhuM02) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb:430]: can we remove kernel spec etc?

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb (hunk 1 above)
Comment (MadhuM02) [reply sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb:316]: never mind, got the diff.

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Score the sample_score.json file using the online endpoint with the azureml endpoint invoke method\n",
+    "response = workspace_ml_client.online_endpoints.invoke(\n",
+    "    endpoint_name=online_endpoint_name,\n",
+    "    deployment_name=demo_deployment.name,\n",
+    "    # demo_deployment.name,\n",
```
Comment (MadhuM02) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb:345]: can remove the commented out line

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
...Agr27HCfsi4Knpc6iev4YLMAv3Rb09vrgTgb7fu/3wjsGlBZNeU6FVi3x7dv8AXGGYXIIII3HrNicClZl3UkHT1H0wlJvHvv16/XFYzUSpI7j1gXIN1t/LDWKqwJFvYnt+HTDn2kP4YyHeQX36YDsFE19Lsok1aR0ud8L5hEei2o3P5YSXqPrjFH9f7YIOCixW0goL3ts1tsPuFQFV03Gpgfb2wIbzyA+2Cy/u/QYYYkI8EORxJ6bWub7n+WBysQLMSCOx98GlAsduxwxd4iTudI3wUhxNE1y6ozKAh2vY9fqMC0pqt1PUgAC2CzbhL9xvgZ/yye9uuLRlVA9aixAQQXbUoF9z1H59cK7tp84IQV6WNgMEsDpuP3BgDE6rX/fGFGITHJZ5mlQWNze9wbBcIzq7XtsWsfjCWuHB/jXCqTc7/wDq/wBsQiqRK8jqQEAZTbphRbzSWk3JsL77YScWC2/iH9cPH3v/AGthBmnoN1IykSEDf/27YXz5Qlo2ANt/brhkR3b8f6YJCB9lJt3OGIoFUcCiQEKNUj32v72wjq5BIsb774HciJrH3xIk/wAkf9AwwRIqkgk0enTe/awwRVPnFu2rrYYDH99P+oYJci9j74BOKIxwSsz6tXe9732OCrICpYGzHt7YEv8A9sT9MOH/ANw4/wCUYUmqIzRUMl1OsAH3a9sJHOC7I0Z3JsBuD89cYn31Hx/YYeoHmDb91f6nFLsyi1IsuyhbBwt+nW+2MVgA2k29diBhknSM/wC+uFb7t/8AnP8AfFJFFYRgi3LSFUYhggtf+uMJYABFNx6bja/c4DGT9o6/uH+hwf8A/wAe/exw1cVKVCZI0iDTcuDqvY4VpVGwBuR1PfrhR1j/ABxku439h/TEKNAsUrJG2rr10f8AfGRyh9KC/Q+nAelWtv4h/XBI+x/5f7YKROKL5zBhqDdLG3b3w6Kd19Tysh121WuQB2GDUgBUXHb+2AIATuMOAC0qOzCeZG0oINI9QP3iSPe39vYYeNIX1yX9ILADYnEVyVFwbdemJEG6m/8A/SX+uBlRAdZPqWXUNJ8q4u2ncj6YZAxv5YNha4F/p+eF6xAn/wDrLgb7Vgt/CMWUqinvHGGurDUt7Bu+FsGYldRsPVc/198JHvYn5/rhw2G3crf+eK3ndUABKGXmv6StvpjMKepxmKd4qL//2Q==",
```
Comment (MadhuM02) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb:250]: please clear out the outputs

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+   "source": [
+    "## Mask Generation Inference using Batch Endpoints\n",
+    "\n",
+    "This sample shows how to deploy `mask-generation` type models to a batch endpoint for inference.\n",
+    "\n",
+    "### Task\n",
+    "`mask-generation` takes in images and prompts (input points, input_boxes, input_labels) and for each image, generates segmentation masks based on the prompts given.\n",
+    "\n",
+    "### Model\n",
+    "Models that can perform the `mask-generation` task are tagged with `mask-generation`. We will use the `facebook/sam-vit-huge` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name. If you don't find a model that suits your scenario or domain, you can discover and [import models from HuggingFace hub](../../import/import_model_into_registry.ipynb) and then use them for inference.\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb:16]:
```suggestion
    "Models that can perform the `mask-generation` task are tagged with `mask-generation`. We will use the `facebook/sam-vit-huge` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name.\n",
```

Comment (rjaincc) [reply sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb:16]: We have not added it to import model yet

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb [hunk 6]
```diff
@@ -0,0 +1,517 @@
+{
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb:1]: Clean cell outputs

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb [hunk 7]
```diff
@@ -0,0 +1,517 @@
...
+    "Models that can perform the `mask-generation` task are tagged with `mask-generation`. We will use the `facebook/sam-vit-huge` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name. If you don't find a model that suits your scenario or domain, you can discover and [import models from HuggingFace hub](../../import/import_model_into_registry.ipynb) and then use them for inference.\n",
+    "\n",
+    "### Inference data\n",
+    "We will use the [odFridgeObjects](https://cvbp-secondary.z19.web.core.windows.net/datasets/object_detection/odFridgeObjects.zip) dataset.\n",
+    "\n",
+    "\n",
+    "### Outline\n",
+    "1. Setup pre-requisites\n",
+    "2. Pick a model to deploy\n",
+    "3. Prepare data for inference - Using a folder of CSV files with base64 images\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb:25]:
```suggestion
    "3. Prepare data for inference - using a folder of CSV files with base64 images\n",
```

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb [hunk 8]
```diff
@@ -0,0 +1,517 @@
...
+    "4. Deploy the model to a batch endpoint\n",
+    "5. Test the endpoint - Using a folder of CSV files with base64 images\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb:27]:
```suggestion
    "5. Test the endpoint - using a folder of CSV files with base64 images\n",
```

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb [hunk 9]
```diff
@@ -0,0 +1,517 @@
...
+    "\n",
+    "endpoint = workspace_ml_client.batch_endpoints.get(endpoint_name)\n",
+    "print(f\"The default deployment is {endpoint.defaults.deployment_name}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 5. Test the endpoint - Using CSV input with base64 images from 3\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb:414]:
```suggestion
    "### 5. Test the endpoint - using CSV input with base64 images from 3\n",
```

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb [hunk 10]
```diff
@@ -0,0 +1,517 @@
...
+    "df.to_csv(batch_input_csv_file_path, index=False)\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4. Deploy the model to a batch endpoint\n",
+    "Batch endpoints are endpoints that are used to do batch inferencing on large volumes of data over a period of time. The endpoints receive pointers to data and run jobs asynchronously to process the data in parallel on compute clusters. Batch endpoints store outputs to a data store for further analysis. For more information on batch endpoints and deployments see [What are batch endpoints?](https://learn.microsoft.com/en-us/azure/machine-learning/concept-endpoints?view=azureml-api-2#what-are-batch-endpoints).\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb:315]:
```suggestion
    "Batch endpoints are endpoints that are used to do batch inferencing on large volumes of data over a period of time. The endpoints receive pointers to data and run jobs asynchronously to process the data in parallel on compute clusters. Batch endpoints store outputs to a data store for further analysis. For more information on batch endpoints and deployments see [What are batch endpoints?](https://learn.microsoft.com/en-us/azure/machine-learning/concept-endpoints?view=azureml-api-2#what-are-batch-endpoints)\n",
```

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb [hunk 11]
```diff
@@ -0,0 +1,517 @@
...
+    ")\n",
+    "workspace_ml_client.begin_create_or_update(endpoint).result()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create a batch deployment"
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb:361]: Please mention the details about min_batch_size. 

You can refer to the notebooks on SD for this.

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb [hunk 12]
```diff
@@ -0,0 +1,449 @@
...
+   "source": [
+    "## Mask Generation Inference using Online Endpoints\n",
+    "\n",
+    "This sample shows how to deploy `mask-generation` type models to an online endpoint for inference.\n",
+    "\n",
+    "### Task\n",
+    "`mask-generation` takes in images and prompts (input points, input_boxes, input_labels) and for each image, generates segmentation masks based on the prompts given.\n",
+    "\n",
+    "### Model\n",
+    "Models that can perform the `mask-generation` task are tagged with `mask-generation`. We will use the `facebook/sam-vit-huge` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name. If you don't find a model that suits your scenario or domain, you can discover and [import models from HuggingFace hub](../../import/import_model_into_registry.ipynb) and then use them for inference.\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb:16]:
```suggestion
    "Models that can perform the `mask-generation` task are tagged with `mask-generation`. We will use the `facebook-sam-vit-huge` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name.\n",
```

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb (hunk 2 above)
Comment (rjaincc) [reply sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb:430]: +1

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb [hunk 13]
```diff
@@ -0,0 +1,449 @@
...
+    ")\n",
+    "\n",
+    "print(f\"raw response: {response}\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Visualize generated masks."
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-online-endpoint.ipynb:356]:
```suggestion
    "#### Visualize generated masks."
```

File: sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb [hunk 14]
```diff
@@ -0,0 +1,517 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/mask-generation/mask-generation-batch-endpoint.ipynb:497]: Clean the metadata

**Comment by MadhuM02** [comment]
make sure to check the notebooks are working end to end before merging.

**Review by MadhuM02 — APPROVED** [review]

**Pull request merged**